pywemo/discovery.py

```diff
@@ -93,15 +93,27 @@ def device_from_uuid_and_location(uuid, mac, location,
     if uuid.startswith('uuid:Humidifier'):
         return Humidifier(url=location, mac=mac,
                           rediscovery_enabled=rediscovery_enabled)
+    if uuid.startswith('uuid:'):
+        try:
+            # unknown device, try to make a basic object to allow manual
+            # interaction
+            return UnknownDevice(url=location, mac=mac,
+                                 rediscovery_enabled=rediscovery_enabled)
+        except (AttributeError, requests.exceptions.RequestException):
```

Comment:
Slightly concerned about this `failing silently`. I recall seeing an error about retrieving `Setup.XML` at some point. It might have been before we checked the device manufacturer - or might have been a crashed WeMo device.

Fine to catch this is you think it's needed - but perhaps put something in the log?

I think by the time we get here - we know it is a Belkin Device - but if that's what you want to check for - perhaps don't return an Unknown device.

Reply:
see my new comment below, but yes we do know this is a Belkin device.  So calling it "unknown" is perhaps a bit of a misnomer, but I just meant that the product is unknown to pywemo.

I caught those two exceptions just because I didn't want some unfamiliar product to cause issues with the continued discovery - perhaps a future product will have a different setup.xml schema.  The `AttributeError` can be raised if a setup.xml file is found but doesn't contain the information needed (for example if it didn't include `friendlyName`).  But I'm happy to just let it raise an exception instead too.